src/Timers.cpp

```diff
@@ -410,8 +859,134 @@ PVR_ERROR Timers::UpdateTimer(const PVR_TIMER &timer)
+    strTmp += StringUtils::Format("&timespanFrom=%02d:%02d", timeinfo.tm_hour, timeinfo.tm_min);
+  }
+
+  if (!timer.bEndAnyTime)
+  {
+    time_t endTime = timer.endTime;
+    std::tm timeinfo = *std::localtime(&endTime);
+    strTmp += StringUtils::Format("&timespanTo=%02d:%02d", timeinfo.tm_hour, timeinfo.tm_min);
+  }
+
+  strTmp += StringUtils::Format("&encoding=%s", vuData.URLEncodeInline(AUTOTIMER_ENCODING).c_str());
+
+  if (timer.bFullTextEpgSearch)
+    strTmp += StringUtils::Format("&searchType=%s", vuData.URLEncodeInline(AUTOTIMER_SEARCH_TYPE_DESCRIPTION).c_str());  
+  else
+    strTmp += StringUtils::Format("&searchType=%s", vuData.URLEncodeInline(AUTOTIMER_SEARCH_TYPE_EXACT).c_str());
+
+  if (!AUTOTIMER_DEFAULT_OMIT(timerToUpdate.searchCase))
+  strTmp += StringUtils::Format("&searchCase=%s", vuData.URLEncodeInline(AUTOTIMER_SEARCH_CASE_SENSITIVE).c_str());
+
+  std::underlying_type<AutoTimer::DeDup>::type deDup = static_cast<AutoTimer::DeDup>(timer.iPreventDuplicateEpisodes);
+  if (deDup == AutoTimer::DeDup::DISABLED)
+  {
+    strTmp += StringUtils::Format("&avoidDuplicateDescription=0");
+  }
+  else
+  {
+    strTmp += StringUtils::Format("&avoidDuplicateDescription=%s", AUTOTIMER_AVOID_DUPLICATE_ANY_SERVICE_OR_RECORDING);
+    switch (deDup)
+    {
+      case AutoTimer::DeDup::CHECK_TITLE:
+        strTmp += StringUtils::Format("&searchForDuplicateDescription=%s", AUTOTIMER_CHECK_SEARCH_FOR_DUP_IN_TITLE);
+        break;
+      case AutoTimer::DeDup::CHECK_TITLE_AND_SHORT_DESC:
+        strTmp += StringUtils::Format("&searchForDuplicateDescription=%s", AUTOTIMER_CHECK_SEARCH_FOR_DUP_IN_TITLE_AND_SHORT_DESC);
+        break;
+      case AutoTimer::DeDup::CHECK_TITLE_AND_ALL_DESCS:
+        //TODO Below is unsupported currently due to bug in the OpenWebIf API, the value cannot be unset
+        //strTmp += StringUtils::Format("&searchForDuplicateDescription=%s", AUTOTIMER_CHECK_SEARCH_FOR_DUP_IN_TITLE_AND_ALL_DESCS);
+        break;
+    }
+  }
+
+  if (timerToUpdate.anyChannel && timer.iClientChannelUid != PVR_TIMER_ANY_CHANNEL)
+  {
+    //move to single channel
+    strTmp += StringUtils::Format("&services=%s", vuData.URLEncodeInline(vuData.GetChannels().at(timer.iClientChannelUid-1).strServiceReference).c_str());
+  }
+  else if (!timerToUpdate.anyChannel && timer.iClientChannelUid == PVR_TIMER_ANY_CHANNEL)
+  {
+    //Move to any channel
+    strTmp += StringUtils::Format("&services=");
+  }
+
+  bool everyday = true;
+  std::string includeParams;
+  if (timer.iWeekdays)
+  {
+      for (int i = 0; i < 7; i++)
+      {
+        if (1 & (timer.iWeekdays >> i))
+        {
+          includeParams += StringUtils::Format("&dayofweek=%d", i);
+        }
+        else
+        {
+          everyday = false;
+        }
+      }
+
+      if (everyday)
+        includeParams = "&dayofweek=";
+  }
+  else
+  {
+    includeParams = "&dayofweek=";
+  }
+  strTmp += includeParams;
```

Comment:
Haven't I seen (large parts of) this code block before? You should consider refactoring to get rid of copied code.

Reply:
`Timers::AddAutoTimer` looks very similar if not (not checked) identical.